### DISCUSSION TITLE
How to get even more precise lineart? Also non-balanced "modes" seem to just be "guess" modes?

### DISCUSSION THREAD

**Discussion by IndolentKheper** [Mikubill/sd-webui-controlnet#1192] [General]
My main goal with Controlnet is to get lineart to transfer 1:1 to a new image, and with full body images it's always been a challenge even at higher resolutions. With close-ups it's fairly easy to get 1:1 lineart, but there's still always some variation even with the strictest of settings. Faces, especially, don't fare well when zoomed out. While I could get around this issue by using an extreme face close-up to clean up the lineart, then copy and paste that onto the less accurate full-body generations, it'd probably be more efficient if it were possible with full-body from the get-go.

The problem for me, I think, has to do with line widths not being fully respected the further back the zoom is.
![demwidtstho](https://user-images.githubusercontent.com/101611653/236665970-9c738833-769b-4fee-b47f-ceff5e8dc6f2.jpg)
Zoomed in, obviously it's a lot easier to get it to transfer line work pretty much 1:1, given that SD has a lot more pixels to focus on those smaller details.  The highest I can push it in img2img for a square image is 1464x1464, not enough for 1:1 line transfer. If I crop to portrait I can generate to the max of 920x2048, but that's still not enough for comparable line work, and actually produces worse lines than I can get at 1440x1440.

There seems to be a tendency, regardless of resolution, models, etc., for lines to be thinned rather than for their widths and shapes to remain intact. Which, for some styles, can be a little destructive when certain lines need to be wider than what SD wants them to be. 

On another note, regarding Control modes, it seems like the mode that most strictly adheres to the Controlnet input image is the "balanced" mode rather than "Controlnet is more important," as the name seems to imply.

![controlmodewhatitdo](https://user-images.githubusercontent.com/101611653/236669995-953ddf7f-ae32-4210-928c-0de399e85e3d.jpg)

The "balanced" outputs are much closer to the Controlnet lineart_standard input image and the img2img input, while "prompt is more important" and "controlnet is more important" both result in "guess" style outputs.

**Comment by lllyasviel** [comment]
a u familiar with PS different blending modes ? perhaps can try blend linearts again in traditional way

**Reply by IndolentKheper** [reply]
Yes, actually that's one of the workarounds I've been using for some of the more stubborn sketches, by superimposing a highly contrasted sketch over itself on a blend layer after having SD color it, then running that through img2img again to get more accurate lines. Also have used Controlnet's preprocessor previews to find stubborn noise, clean up the preview, then invert it and use it as a blending layer or run it back through lineart_standard. 

Another trick I figured out is to shrink a lineart image then upscale it in SD, which can clean up a lot of noise without much loss of detail. Something like this:
![shrinkup](https://user-images.githubusercontent.com/101611653/236679466-fcd3c255-db49-4bf2-9b8b-e44b772ab313.jpg)
A lot of noise is lost when shrinking, then the upscale has less noise to recreate. A lot cleaner, though there is a small loss of detail and line width, but overall better for use in Controlnet.
